components/ai_chat/resources/page/components/main/index.tsx

```diff
@@ -104,10 +107,57 @@ function Main() {
     }
   }
 
+  let inputFocused: boolean = false
+  let viewPortWithoutKeyboard = 0
+  let keyboardSize = 0
```

Comment:
These won't persist between rerenders of the component, so they can get out of sync with the input's actual state.

```suggestion
  const inputFocused = useRef(false)
  const viewPortWithoutKeyboard = useRef(0)
  const keyboardSize = useRef(0)
```